[ENH] BEP030: Functional Near-Infrared Spectroscopy (#802)
bids-standard/bids-specification: src/04-modality-specific-files/10-functional-near-infrared-spectroscopy.md

# Functional Near-Infrared Spectroscopy

Support for functional Near-Infrared Spectroscopy (fNIRS) was developed as a
[BIDS Extension Proposal](../07-extensions.md#bids-extension-proposals).
Please see [Citing BIDS](../01-introduction.md#citing-bids)
on how to appropriately credit this extension when referring to it in the
context of the academic literature.

## fNIRS recording data

{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["nirs", "events"]) }}

> Only the Shared Near Infrared File Format ([SNIRF](https://github.com/fNIRS/snirf))
> format is supported in BIDS. The SNIRF

Reviewer: ```suggestion
Only the Shared Near Infrared Spectroscopy Format ([SNIRF](https://github.com/fNIRS/snirf))
file specification is supported in BIDS. The SNIRF
```

specification supports one or more fNIRS datasets to be stored in a single

> `.snirf` file. However, to be BIDS compatible each SNIRF file MUST contain

Reviewer: ```suggestion
`.snirf` file. However, to be BIDS compatible, each SNIRF file MUST contain
```

only a single run. A limited set of fields from the SNIRF specification are

> replicated  in the BIDS specification. This redundancy allows the data to be
> easily parsed by humans and machines that do not have a SNIRF reader at hand,
> which improves findability and tooling development.

Reviewer: ```suggestion
replicated  in the BIDS specification. This redundancy allows the data to be
easily parsed by humans and machines that do not have a SNIRF reader at hand,
which improves findability and tooling development.
```

> Original unprocessed fNIRS data in the native format, if different, can also

Reviewer: "if different" from what?

be stored in the [`/sourcedata` directory](../02-common-principles.md#source-vs-raw-vs-derived-data)
along with code to convert the data to
SNIRF in the [`/code` directory](../02-common-principles.md#storage-of-derived-datasets).

> The unprocessed raw data should be stored in
> the manufacturers format before any additional processing or conversion.
> The native file format is especially valuable in case conversion elicits the
> loss of crucial metadata specific to manufacturers and specific fNIRS systems.

Reviewer: ```suggestion
The unprocessed raw data should be stored in
the manufacturer's format before any additional processing or conversion is applied.
Retaining the native file format is especially valuable in a case when conversion elicits the
loss of crucial metadata unique to manufacturers and specific fNIRS systems.
```

We also encourage users to provide additional meta information extracted from

> the manufacturer specific data files in the sidecar JSON file. This allows for

Reviewer: ```suggestion
the manufacturer-specific data files in the sidecar JSON file. This allows for
```

easy searching and indexing of key metadata elements without needing to parse
the various proprietary (typically binary) native data files. Other relevant
files should  be included alongside the fNIRS data.

### Terminology

> For proper documentation of fNIRS recording metadata it is important
> to understand the difference between a source, detector, and channel as these
> are defined differently to other modalities such as EEG. The following definitions

Reviewer: ```suggestion
For proper documentation of fNIRS recording metadata, it is important
to understand the difference between a Source, Detector, and Channel as these
are defined differently to other modalities, such as EEG. The following definitions
```

apply in this document:

-   Source - A light emitting device, sometimes called a transmitter.

-   Detector - A photoelectric transducer, sometimes called a receiver.

> -   Channel - A paired coupling of a source and a detector with one specific light wavelength.

Reviewer: ```suggestion
-   Channel - A paired coupling of a source and a detector with one specific wavelength of light.
```

    It is common for a single Source-Detector pair to result in two or more channels

>     with different wavelengths.

Reviewer: Do you also want to define "Optode" here as well? You use the term a lot later on in the document. "Optode can refer to either a source or a detector" etc etc

Reviewer 2: This terminology section is very helpful, thanks!!

### Sidecar JSON (`*_nirs.json`)

It is common within the fNIRS community for researchers to build their own caps
and optode holders to position their sources and detectors, or for optodes to
be directly attached to the scalp with adhesive. To facilitate description of

> the wide variety of possible configurations several fields are RECOMMENDED within

Reviewer: ```suggestion
the wide variety of possible configurations, several fields are RECOMMENDED within
```

the `_*nirs.json` file.

> To clarify the usage and interaction of these fields the following examples are provided.

Reviewer: ```suggestion
To clarify the usage and interaction of these fields, the following examples are provided.
```

If a commercial cap such as EasyCap actiCAP 64 Ch Standard-2

> was used then the values `CapManufacturer = ”EasyCap”`, `CapManufacturersModelName = ”actiCAP

Reviewer: ```suggestion
was used, then the values `CapManufacturer = ”EasyCap”`, `CapManufacturersModelName = ”actiCAP
```

64 Ch Standard-2”` and `NIRSPlacementScheme = ”n/a”` should be used.
If an EasyCap was used but with custom positions,
as may be done by cutting custom holes in the cap,
then the values `CapManufacturer = ”EasyCap”`, `CapManufacturersModelName = ”custom”` and
`NIRSPlacementScheme = ”n/a”` should be used.

> If a completely custom cap was knitted then

Reviewer: ```suggestion
If a completely custom cap was knitted, then
```

`CapManufacturer = ”custom”`, `CapManufacturersModelName = ”custom”` and
`NIRSPlacementScheme = ”n/a”`.
If no cap was used and optodes were taped to the scalp
at positions Cz, C1 and C2, then the values `CapManufacturer = ”none”`, `CapManufacturersModelName
= ”n/a”` and `NIRSPlacementScheme = ”["Cz", "C1", “C2]”` should be used.

> Closely spaced source detector pairs are often included in fNIRS measurements to

Reviewer: ```suggestion
Closely spaced or short-separation source-detector pairs are often included in fNIRS measurements to
```

obtain a measure of systemic, rather than neural, activity. These source-detector
pairs are referred to as short channels. There is variation in how manufacturers
implement these short channels, some use specialised sources or detectors,

> and the placement mechanisms vary. The exact specifications on the type of

Reviewer: ```suggestion
and the placement mechanisms vary. The exact specifications of the type of
```

specialised short channel optodes is beyond the scope of the BIDS specification

> and this information can be stored within the SNIRF file (for example, in the `sourcePower` field).

Reviewer: This wording is odd. "It's beyond the scope of the BIDS specs _and_ this info can be stored..." Could you say that it's beyond the scope of the BIDS specs but the info should still be stored within the SNIRF file?

Author: I have reworded. Hopefully this is clearer now.

> However, to improve searchability and ease of access for users it is useful to
> know if short channels were included in the fNIRS measurements, this information
> is stored in the field `ShortChannelCount`.

Reviewer: ```suggestion
However, to improve searchability and ease of access for users, it is useful to
know if short channels were included in the fNIRS measurements; this information
is stored in the field `ShortChannelCount`.
```

If the field `ShortChannelCount` is populated, then the optional column `short_channel`
may be used in `*_channels.tsv` to describe which channels were specified as short.

Generic fields that MUST be present:

| **Key name** | **Requirement level** | **Data type** | **Description**                                                                                                                                                                                                                                                                                                                                                                    |
| ------------ | --------------------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | TaskName     | REQUIRED              | [string][]    | Name of the task. No two tasks should have the same name. The task label included in the file name is derived from this TaskName field by removing all non-alphanumeric (`[a-zA-Z0-9]`) characters. For example `TaskName` `"faces n-back"` will correspond to task label `facesnback`. A RECOMMENDED convention is to name resting state task using labels beginning with `rest`. |

Reviewer: Why can't two tasks have the same name? What if they were the same task? Does this mean that two separate tasks within the same file can't have the same name? I think another sentence is needed to clarify what you mean.

Author: You cant have two seperate tasks in one file. I think the common understanding of task and the definition differ here, causing this confusion. I paste the definition from BIDS below.

Task - a set of structured activities performed by the participant. Tasks are usually accompanied by stimuli and responses, and can greatly vary in complexity. For the purpose of this specification we consider the so-called "resting state" a task. In the context of brain scanning, a task is always tied to one data acquisition. Therefore, even if during one acquisition the subject performed multiple conceptually different behaviors (with different sets of instructions) they will be considered one (combined) task.

Generic fields which SHOULD be present: For consistency between studies and institutions, we encourage users to extract the values of these fields from the actual raw data. Whenever possible, please avoid using ad hoc wording.

| **Key name**               | **Requirement level** | **Data type**                        | **Description**                                                                                                                                                                                                                                                                                                                        |
| -------------------------- | --------------------- | ------------------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | InstitutionName            | RECOMMENDED           | [string][]                           | The name of the institution in charge of the equipment that produced the composite instances.                                                                                                                                                                                                                                          |

Reviewer: What's a "composite instance"? Same comment for the Description column in the next row.

Author: Apparently its an fMRI thing. Who knew. I should change it. And perhaps throughout the rest of BIDS.

https://www.medicalconnections.co.uk/kb/Composite-Instance

| InstitutionAddress         | RECOMMENDED           | [string][]                           | The address of the institution in charge of the equipment that produced the composite instances.                                                                                                                                                                                                                                       |
| Manufacturer               | RECOMMENDED           | [string][]                           | Manufacturer of the fNIRS system (for example, `"NIRx"`, `"Artinis"`, `"Hitachi"`).                                                                                                                                                                                                                                                    |

> | ManufacturersModelName     | RECOMMENDED           | [string][]                           | Manufacturer's designation of the EEG system model (for example, `"NIRScout"`).                                                                                                                                                                                                                                                        |

Reviewer: Do you need to replace "EEG" with "fNIRS" here?

| SoftwareVersions           | RECOMMENDED           | [string][]                           | Manufacturer's designation of the acquisition software.                                                                                                                                                                                                                                                                                |
| TaskDescription            | RECOMMENDED           | [string][]                           | Description of the task.                                                                                                                                                                                                                                                                                                               |

> | Instructions               | RECOMMENDED           | [string][]                           | Text of the instructions given to participants before the scan. This is not only important for behavioral or cognitive tasks but also in resting state paradigms (for example, to distinguish between eyes open and eyes closed).                                                                                                      |

Reviewer: ```suggestion
| Instructions               | RECOMMENDED           | [string][]                           | Text of the instructions given to participants before the measurement. This is not only important for behavioral or cognitive tasks but also in resting state paradigms (for example, to distinguish between eyes open and eyes closed).                                                                                                      |
```

| RecordingDuration          | RECOMMENDED           | [number][]                           | Length of the recording in seconds (for example, 3600).                                                                                                                                                                                                                                                                                |
| HeadCircumference          | RECOMMENDED           | [number][]                           | Circumference of the participants head, expressed in cm (for example, 58).                                                                                                                                                                                                                                                             |

> | CapManufacturer            | RECOMMENDED           | [string][]                           | Name of the cap manufacturer (for example, "Artinis") If a custom made cap is used then the string “custom” should be used. If no cap was used, such as with optodes that are directly taped to the scalp, then the string “none” should be used and NIRSPlacementScheme field may be used to specify the optode placement scheme.     |

Reviewer: ```suggestion
| CapManufacturer            | RECOMMENDED           | [string][]                           | Name of the cap manufacturer (for example, "Artinis") If a custom-made cap is used then the string “custom” should be used. If no cap was used, such as with optodes that are directly taped to the scalp, then the string “none” should be used and NIRSPlacementScheme field may be used to specify the optode placement scheme.     |
```

> | CapManufacturersModelName  | RECOMMENDED           | [string][]                           | Manufacturer's designation of the fNIRS cap model (for example, "Headband with print (S-M)"). If a cap from a standard manufacturer was modified then the field should be set to “custom” If no cap was used then the CapManafacturer field should be “none” and this field should be “n/a”                                            |

Reviewer: ```suggestion
| CapManufacturersModelName  | RECOMMENDED           | [string][]                           | Manufacturer's designation of the fNIRS cap model (for example, "Headband with print (S-M)"). If a cap from a standard manufacturer was modified, then the field should be set to “custom.” If no cap was used, then the CapManafacturer field should be “none” and this field should be “n/a.”                                            |
```

| HardwareFilters            | RECOMMENDED           | [object][] of [objects][] or `"n/a"` | [Object][] of temporal hardware filters applied, or `"n/a"` if the data is not available. Each key:value pair in the JSON object is a name of the filter and an object in which its parameters are defined as key:value pairs. For example, `{"Highpass RC filter": {"Half amplitude cutoff (Hz)": 0.0159, "Roll-off": "6dB/Octave"}}` |
| SubjectArtefactDescription | RECOMMENDED           | [string][]                           | Free-form description of the observed subject artifact and its possible cause (for example, "Vagus Nerve Stimulator", "non-removable implant"). If this field is set to `n/a`, it will be interpreted as absence of major source of artifacts except cardiac and blinks.                                                               |

> Specific fNIRS specific fields that MUST be present:

Reviewer: ```suggestion
Specific fNIRS fields that MUST be present:
```

| **Key name**            | **Requirement level**                                | **Data type**     | **Description**                                                                                                                                                                                                                                                                                       |
| ----------------------- | ---------------------------------------------------- | ----------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | SamplingFrequency       | REQUIRED                                             | [number][] or n/a | Sampling frequency (in Hz) of all the data in the recording, regardless of their type (for example, 12).  If individual channels have different sampling rates then the field here should be specified as “n/a” and the values  should be specified in the sampling_frequency column in channels.tsv. |

Reviewer: ```suggestion
| SamplingFrequency       | REQUIRED                                             | [number][] or n/a | Sampling frequency (in Hz) of all the data in the recording, regardless of their type (for example, 12).  If individual channels have different sampling rates, then the field here should be specified as “n/a” and the values  should be specified in the sampling_frequency column in channels.tsv. |
```

| NIRSChannelCount        | REQUIRED                                             | [number][]        | Number of fNIRS channels.                                                                                                                                                                                                                                                                             |
| NIRSSourceOptodeCount   | REQUIRED                                             | [number][]        | Number of fNIRS sources.                                                                                                                                                                                                                                                                              |
| NIRSDetectorOptodeCount | REQUIRED                                             | [number][]        | Number of fNIRS detectors.                                                                                                                                                                                                                                                                            |

> | ACCELChannelCount       | RECOMMENDED but REQUIRED if any channel type is ACC  | [number][]        | Number of accelerometer channels                                                                                                                                                                                                                                                                      |
> | GYROChannelCount        | RECOMMENDED but REQUIRED if any channel type is GYRO | [number][]        | Number of gyrometer channels                                                                                                                                                                                                                                                                          |
> | MAGNChannelCount        | RECOMMENDED but REQUIRED if any channel type is MAGN | [number][]        | Number of magnetometer channels                                                                                                                                                                                                                                                                       |

Reviewer: ```suggestion
| ACCELChannelCount       | RECOMMENDED but REQUIRED if any channel type is ACC  | [number][]        | Number of accelerometer channels.                                                                                                                                                                                                                                                                     |
| GYROChannelCount        | RECOMMENDED but REQUIRED if any channel type is GYRO | [number][]        | Number of gyrometer channels.                                                                                                                                                                                                                                                                         |
| MAGNChannelCount        | RECOMMENDED but REQUIRED if any channel type is MAGN | [number][]        | Number of magnetometer channels.                                                                                                                                                                                                                                                                       |
```

Specific fNIRS fields that SHOULD be present:

| **Key name**        | **Requirement level** | **Data type**         | **Description**                                                                                                                                                                                                                                                                                                                                                                       |
| ------------------- | --------------------- | --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | SourceType          | RECOMMENDED           | [number][] or `mixed` | Type of the source. Preferably a specific model/part number is supplied. This is a freeform description. But the following keywords are suggested. LED, LASER, VCSEL. If individual channels have different SourceTypes then the field here should be specified as `mixed`  and this column should be included in `optodes.tsv`.                                                      |

Reviewer: ```suggestion
| SourceType          | RECOMMENDED           | [number][] or `mixed` | Type of source. Preferably a specific model/part number is supplied. This is a freeform description, but the following keywords are suggested: LED, LASER, VCSEL. If individual channels have different SourceTypes, then the field here should be specified as `mixed`  and this column should be included in `optodes.tsv`.                                                      |
```

> | DetectorType        | RECOMMENDED           | [number][] or `mixed` | Type of the detector. This is a free form description with the following suggested terms SiPD, APD. Preferably a specific model/part number is supplied. If individual channels have different DetectorTypes then the field here should be specified as `mixed`  and this column should be included in `optodes.tsv`.                                                                 |

Reviewer: ```suggestion
| DetectorType        | RECOMMENDED           | [number][] or `mixed` | Type of detector. This is a free form description with the following suggested terms: SiPD, APD. Preferably a specific model/part number is supplied. If individual channels have different DetectorTypes, then the field here should be specified as `mixed`  and this column should be included in `optodes.tsv`.                                                                 |
```

| ShortChannelCount   | RECOMMENDED           | [number][]            | The number of short channels. 0 indicates no short channels.                                                                                                                                                                                                                                                                                                                          |

> | NIRSPlacementScheme | RECOMMENDED           | [string][]            | Placement scheme of NIRS optodes. Either the name of a standardized placement system (for example, "10-20") or a list of standardized position names (for example, ["Cz", "Pz"]). This field should only be used if a cap was not used. If a standard cap was used then it should be specified in CapManufacturer and CapManufacturersModelName and this field should be set to “n/a” |

Reviewer: ```suggestion
| NIRSPlacementScheme | RECOMMENDED           | [string][]            | Placement scheme of NIRS optodes. Either the name of a standardized placement system (for example, "10-20") or a list of standardized position names (for example, ["Cz", "Pz"]). This field should only be used if a cap was not used. If a standard cap was used, then it should be specified in CapManufacturer and CapManufacturersModelName and this field should be set to “n/a” |
```

Example:

```JSON
{
  "TaskName":"visual",
  "InstitutionName":"Macquarie University. Australian Hearing Hub",
  "InstitutionAddress":"6 University Ave, Macquarie University NSW 2109 Australia",
  "Manufacturer":"NIRx",
  "ManufacturersModelName":"NIRScout",
  "TaskDescription":"visual gratings and noise patterns",
  "Instructions":"look at the dot in the center of the screen and press the button when it changes color",
  "SamplingFrequency":3.7,
  "NIRSChannelCount":56,
  "NIRSSourceOptodeCount":16,
  "NIRSDetectorOptodeCount":16,
  "ACCELChannelCount":0,
  "SoftwareFilters":"n/a",
  "RecordingDuration":233.639,
  "DCOffsetCorrection":0,
  "HardwareFilters":{"Highpass RC filter": {"Half amplitude cutoff (Hz)": 0.0159, "Roll-off": "6dBOctave"}},
  "CapManafacturer":"NIRx",
  "CapManufacturersModelName":"Headband with print (S-M)",
}
```

> ### Participants file

Reviewer: ```suggestion
### Participant file
```

> For fNIRS data the modality agnostic participant.tsv and json files SHOULD contain

Reviewer: ```suggestion
For fNIRS data, the modality agnostic participant.tsv and json files SHOULD contain
```

> For fNIRS data the modality agnostic participant.tsv and json files SHOULD contain
> the age field, this is required for calculation of age specific path length factors.

Reviewer: ```suggestion
For fNIRS data, the modality agnostic participant.tsv and json files SHOULD contain
the age field, as this is required for calculation of age-specific pathlength factors.
```

If the dataset contains multiple groups, for example patients and controls, then
a column labeled `group` MAY be included (see [participants.tsv](https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#participants-file)).

```text
participant_id	age	 sex	hand  group
sub-01       	34	 M	    n/a   patient
sub-02	        32	 F	    n/a   control
```

## Channels description (`*_channels.tsv`)

{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["channels"]) }}

> Channels are a pairing of source and detector optodes with a specific light wavelength.

Reviewer: ```suggestion
Channels are a pairing of source and detector optodes with a specific wavelength of light.
```

Unlike in other modalities, not all pairings of optodes correspond to meaningful data
and not all pairs have to be recorded or represented in the data.  Note that the source
and detector names used in the channel specifications are specified in the `*_optodes.tsv`

> file below. The order of the required columns in the `*_channels.tsv` file MUST be listed as below.

Reviewer: Do you mean that the required columns must be listed _in the same order_ as below?

The BIDS specification supports several types of NIRS devices which output raw data in
different forms. The type of measurement is specified in the type column. For example,
when measurements are taken with a continuous wave (CW) device that saves the data

> as optical density, the type NIRSCWOPTICALDENSITY is used with the units unitless,

Reviewer: I think there's a typo here. "...is used with the units unitless" Not sure how to fix it.

| wavelength_actual          | OPTIONAL              | [number][]    | Measured wavelength of light in nm. `n/a` for channels that do not contain raw NIRS signals (acceleration). This field is equivalent to `measurementList.wavelengthActual` in the SNIRF specification.                  |
| description                | OPTIONAL              | [string][]    | Free-form text description of the channel, or other information of interest.                                                                                                                                            |
| wavelength_emission_actual | OPTIONAL              | [number][]    | Measured emission wavelength of light in nm. `n/a` for channels that do not contain raw NIRS signals (acceleration). This field is equivalent to `measurementList.wavelengthEmissionActual` in the SNIRF specification. |
| short_channel              | OPTIONAL              | [boolean][]   | Is the channel designated as short. The total number of channels listed as short channels should be stored in `ShortChannelCount` in `*_fnirs.csv`.                                                                     |

> Restricted keyword list for the channel types.

Reviewer: Do you want to make this line more of a heading? Rather than just a sentence at the start of a paragraph.

All fNIRS channels types MUST correspond to a [valid SNIRF data type](https://github.com/fNIRS/snirf/blob/master/snirf_specification.md#appendix).
Additional channels that are recorded simultaneously with the fNIRS
device and stored in the same data file should be included as well.
However, additional channels that are  simultaneously recorded with a different device
should be stored according to their appropriate modality specification.
For example, motion data that was simultaneously recorded with a different device should be specified
according to BEP029 and not according to the fNIRS data type.

> Whereas, if the motion data was acquired in with the fNIRS device itself it should be included here with the fNIRS data.

Reviewer: ```suggestion
Whereas, if the motion data was acquired in with the fNIRS device itself, it should be included here with the fNIRS data.
```

Any of the channel types defined in other BIDS specification can be used here as well,
and several of these data types which are commonly acquired using fNIRS devices are included at the base of the table.
Note that upper-case is REQUIRED.

| **Keyword**                 | **Description**                                                                                                                                                             |
| --------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| NIRSCWAMPLITUDE             | Continuous wave amplitude measurements. Equivalent to dataType 001 in SNIRF.                                                                                                |
| NIRSCWFLUORESCENSEAMPLITUDE | Continuous wave fluorescence amplitude measurements. Equivalent to dataType 051 in SNIRF.                                                                                   |

> | NIRSCWOPTICALDENSITY        | Gcontinuous wave change in optical density measurements. Equivalent to dataTypeLabel dOD in SNIRF.                                                                          |
> | NIRSCWHBO                   | continuous wave oxygenated hemoglobin (oxyhemoglobin) concentration measurements. Equivalent to dataTypeLabel HbO in SNIRF.                                                 |
> | NIRSCWHBR                   | continuous wave deoxygenated hemoglobin (deoxyhemoglobin) concentration measurements. Equivalent to dataTypeLabel HbR in SNIRF.                                             |
> | NIRSCWMUA                   | continuous wave optical absorption  measurements. Equivalent to dataTypeLabel mua in SNIRF.                                                                                 |

Reviewer: ```suggestion
| NIRSCWOPTICALDENSITY        | Continuous wave change in optical density measurements. Equivalent to dataTypeLabel dOD in SNIRF.                                                                          |
| NIRSCWHBO                   | Continuous wave oxygenated hemoglobin (oxyhemoglobin) concentration measurements. Equivalent to dataTypeLabel HbO in SNIRF.                                                 |
| NIRSCWHBR                   | Continuous wave deoxygenated hemoglobin (deoxyhemoglobin) concentration measurements. Equivalent to dataTypeLabel HbR in SNIRF.                                             |
| NIRSCWMUA                   | Continuous wave optical absorption  measurements. Equivalent to dataTypeLabel mua in SNIRF.                                                                                 |
```

> | ACCEL                       | Accelerometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z) |
> | GYRO                        | Gyrometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z)     |
> | MAGN                        | Magnetomenter channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z) |

Reviewer: ```suggestion
| ACCEL                       | Accelerometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
| GYRO                        | Gyrometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z).     |
| MAGN                        | Magnetomenter channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
```

| MISC                        | Miscellaneous                                                                                                                                                               |

> ```Text

Reviewer: Missing an "Example:" heading before this box

Name         type                   source      detector      wavelength_nominal   units
S1-D1        NIRSCWAMPLITUDE        A1          Fz            760                  V
S1-D1        NIRSCWAMPLITUDE        A1          Fz            850                  V
S1-D2        NIRSCWAMPLITUDE        A1          Cz            760                  V
S2-D1        NIRSCWAMPLITUDE        A2          Fz            760                  V
```

## Optode description (`*_optodes.tsv`)

{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["optodes"]) }}

File that provides the location and type of optodes. Note that coordinates MUST be
expressed  in Cartesian coordinates according to the NIRSCoordinateSystem and
NIRSCoordinateSystemUnits fields in `*_coordsystem.json`. If an `*_optodes.tsv`
file is specified, a `*_coordsystem.json` file MUST be specified as well.
The order of the required columns in the `*_optodes.tsv` file MUST be as listed below.

> The x, y, and z positions are for measured locations, for example with a polhemus

Reviewer: ```suggestion
The x, y, and z positions are for measured locations, for example, with a polhemus
```

digitizer. If you also have idealised positions, where you wish the optodes to be
placed, these can be listed in the template values. SNIRF contains arrays for both

> the 3D and 2D locations of data, the BIDS format MUST store the 3D locations if

Reviewer: ```suggestion
the 3D and 2D locations of data. The BIDS format MUST store the 3D locations if
```

Reviewer: This is a bit confusing. Do you need to specify 3D locations in SNIRF or just in BIDS? What's the difference between these two formats for this context?

The following columns MAY be present:

| **Column name** | **Requirement level**               | **Data type**       | **Definition**                                                                                       |
| --------------- | ----------------------------------- | ------------------- | ---------------------------------------------------------------------------------------------------- |
| template_x      | OPTIONAL but REQUIRED if x is `n/a` | [number][] or `n/a` | Assumed or ideal position along the x axis.                                                          |
| template_y      | OPTIONAL but REQUIRED if x is `n/a` | [number][] or `n/a` | Assumed or ideal position along the y axis.                                                          |
| template_z      | OPTIONAL but REQUIRED if x is `n/a` | [number][] or `n/a` | Assumed or ideal position along the z axis.                                                          |
| description     | OPTIONAL.                           | [string][]          | Free-form text description of the optode, or other information of interest.                          |

> | source_type     | OPTIONAL.                           | [string][]          | The type of detector. Only to be used if the field `DetectorType` in `*_nirs.json` is set to `mixed` |

Reviewer: ```suggestion
| source_type     | OPTIONAL.                           | [string][]          | The type of source. Only to be used if the field `DetectorType` in `*_nirs.json` is set to `mixed` |
```

> | source_type     | OPTIONAL.                           | [string][]          | The type of detector. Only to be used if the field `DetectorType` in `*_nirs.json` is set to `mixed` |
> | detector_type   | OPTIONAL.                           | [string][]          | The type of detector. Only to be used if the field `SourceType` in `*_nirs.json` is set to `mixed`   |

Reviewer: ```suggestion
| detector_type     | OPTIONAL.                           | [string][]          | The type of detector. Only to be used if the field `DetectorType` in `*_nirs.json` is set to `mixed` |
| source_type   | OPTIONAL.                           | [string][]          | The type of source. Only to be used if the field `SourceType` in `*_nirs.json` is set to `mixed`   |
```

Example:

```Text
name    type         x          y         z          template_x    template_y   template_z
A1      source       -0.0707    0.0000    -0.0707    -0.07         0.00         0.07
Fz      detector     0.0000     0.0714    0.0699     0.0           0.07         0.07
S1      source       -0.2707    0.0200    -0.1707    -0.03         0.02         -0.2
D2      detector     0.0022     0.1214    0.0299     0.0           0.12         0.03
```

## Coordinate System JSON (`*_coordsystem.json`)

{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["coordsystem"]) }}

A `*_coordsystem.json` file is used to specify the fiducials, the location of anatomical landmarks,

> and the coordinate system and units in which the position of electrodes and landmarks is expressed.

Reviewer: ```suggestion
and the coordinate system and units in which the position of optodes and landmarks is expressed.
```

> A `*_coordsystem.json` file is used to specify the fiducials, the location of anatomical landmarks,
> and the coordinate system and units in which the position of electrodes and landmarks is expressed.
> Ficiduals are objects with a well defined location used to facilitate the localization of sensors
> and co-registration, anatomical landmarks are locations on a research subject such as the nasion

Reviewer: ```suggestion
A `*_coordsystem.json` file is used to specify the fiducials, the location of anatomical landmarks,
and the coordinate system and units in which the position of optodes and landmarks is expressed.
Ficiduals are objects with a well-defined location used to facilitate the localization of sensors
and co-registration, anatomical landmarks are locations on a research subject such as the nasion
```

Reviewer: Also, is "sensors" the right word here?

(for a detailed definition see the coordinate system description in the BIDS specification).
The `*_coordsystem.json` is REQUIRED if the optional `*_optodes.tsv` is specified. If a corresponding
anatomical MRI is available, the locations of anatomical landmarks in that scan should also be stored

> in the `*_T1w.json` file which goes alongside the MRI data.

Reviewer: ```suggestion
in the `*_T1w.json` file which goes alongside the fNIRS data.
```

Reviewer: Not sure if this should be MRI or nirs

> Not all fnirs systems provide 3D coordinate information or digitisation capabilities. In this case the only x and y are specified and z is n/a

Reviewer: ```suggestion
Not all fNIRS systems provide 3D coordinate information or digitisation capabilities. In this case, the only x and y are specified and z is "n/a"
```

Fields relating to the fNIRS optode positions:

| **Key name**                        | **Requirement level**                                          | **Data type** | **Description**                                                                                                                                                                                                                                                                                                                                            |
| ----------------------------------- | -------------------------------------------------------------- | ------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | NIRSCoordinateSystem                | REQUIRED                                                       | [string][]    | Defines the coordinate system in which  the optode positions are expressed. See Appendix VIII for a list of restricted keywords for coordinate systems. If Other, provide definition of the coordinate system in NIRSCoordinateSystemDescription.                                                                                                          |

Reviewer: ```suggestion
| NIRSCoordinateSystem                | REQUIRED                                                       | [string][]    | Defines the coordinate system in which  the optode positions are expressed. See Appendix VIII for a list of restricted keywords for coordinate systems. If '"Other"', provide definition of the coordinate system in NIRSCoordinateSystemDescription.                                                                                                          |
```

| NIRSCoordinateUnits                 | REQUIRED                                                       | [string][]    | Units in which the coordinates that are listed in the field NIRSCoordinateSystem are represented. MUST be m, cm, or mm.                                                                                                                                                                                                                                    |

> | NIRSCoordinateSystemDescription     | RECOMMENDED, but REQUIRED if `NIRSCoordinateSystem` is `Other` | [string][]    | Freeform text description or link to document describing the NIRS coordinate system system in detail (for example, "Coordinate system with the origin at anterior commissure (AC), negative y-axis going through the posterior commissure (PC), z-axis going to a mid-hemisperic point which lies superior to the AC-PC line, x-axis going to the right"). |

Reviewer: ```suggestion
| NIRSCoordinateSystemDescription     | RECOMMENDED, but REQUIRED if `NIRSCoordinateSystem` is `Other` | [string][]    | Freeform text description or link to document describing the NIRS coordinate system in detail (for example, "Coordinate system with the origin at anterior commissure (AC), negative y-axis going through the posterior commissure (PC), z-axis going to a mid-hemisperic point which lies superior to the AC-PC line, x-axis going to the right"). |
```

> | NIRSCoordinateProcessingDescription | RECOMMENDED                                                    | [string][]    | Has any post-processing (such as projection) been done on the optode positions (for example, "surface_projection", "none").                                                                                                                                                                                                                                |

Reviewer: ```suggestion
| NIRSCoordinateProcessingDescription | RECOMMENDED                                                    | [string][]    | Free-form text description of any post-processing (such as projection) that has been done on the optode positions (for example, "surface_projection", "none").                                                                                                                                                                                                                                |
```

> Fields relating to the position of fiducials measured during an fnirs session/run:

Reviewer: ```suggestion
Fields relating to the position of fiducials measured during an fNIRS session/run:
```

| **Key name**                         | **Requirement level**                                                 | **Data type**            | **Description**                                                                                                                                                                                                                                                                                                                                                 |
| ------------------------------------ | --------------------------------------------------------------------- | ------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | FiducialsDescription                 | OPTIONAL                                                              | [string][]               | Free-form text description of how the fiducials such as vitamin-E capsules were placed relative to anatomical landmarks, and how the position of the fiducials were measured (for example, both with Polhemus and with T1w MRI).                                                                                                                                |

Reviewer: ```suggestion
| FiducialsDescription                 | OPTIONAL                                                              | [string][]               | Free-form text description of how the fiducials, such as "vitamin-E capsules were placed relative to anatomical landmarks," and how the position of the fiducials were measured (for example, both with Polhemus and with T1w MRI).                                                                                                                                |
```

> | FiducialsCoordinates                 | RECOMMENDED                                                           | [object][] of [arrays][] | Key:value pairs of the labels and 3-D digitized position of anatomical landmarks, interpreted following the `FiducialsCoordinateSystem` (for example, `{"NAS": [12.7,21.3,13.9], "LPA": [5.2,11.3,9.6], "RPA": [20.2,11.3,9.1]}`). Each array MUST contain three numeric values corresponding to x, y, and z axis of the coordinate system in that exact order. |

Reviewer: Should "Key:value" have a colon in it?

Author: I think this is correct as is. I checked against the other modalities.

| FiducialsCoordinateUnits             | RECOMMENDED                                                           | [string][]               | Units in which the coordinates that are  listed in the field `FiducialsCoordinateSystem` are represented. MUST be `"m"`, `"cm"`, or `"mm"`.                                                                                                                                                                                                                     |
| FiducialsCoordinateSystemDescription | RECOMMENDED, but REQUIRED if `FiducialsCoordinateSystem` is `"Other"` | [string][]               | Free-form text description of the coordinate system. May also include a link to a documentation page or paper describing the system in greater detail.                                                                                                                                                                                                          |

> Fields relating to the position of anatomical landmark measured during an fnirs session/run:

Reviewer: ```suggestion
Fields relating to the position of anatomical landmarks measured during an fNIRS session/run:
```

| **Key name**                                  | **Requirement level**                                                          | **Data type**            | **Description**                                                                                                                                                                                                                                                                                                                                                          |
| --------------------------------------------- | ------------------------------------------------------------------------------ | ------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |

> | AnatomicalLandmarkCoordinates                 | RECOMMENDED                                                                    | [object][] of [arrays][] | Key:value pairs of the labels and 3-D digitized position of anatomical landmarks, interpreted following the `AnatomicalLandmarkCoordinateSystem` (for example, `{"NAS": [12.7,21.3,13.9], "LPA": [5.2,11.3,9.6], "RPA": [20.2,11.3,9.1]}`). Each array MUST contain three numeric values corresponding to x, y, and z axis of the coordinate system in that exact order. |

Reviewer: "Key:value" correct?

# Functional Near-Infrared Spectroscopy

Support for functional Near-Infrared Spectroscopy (fNIRS) was developed as a
[BIDS Extension Proposal](../07-extensions.md#bids-extension-proposals).
Please see [Citing BIDS](../01-introduction.md#citing-bids)
on how to appropriately credit this extension when referring to it in the
context of the academic literature.

## fNIRS recording data

{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["nirs", "events"]) }}

Only the Shared Near Infrared Spectroscopy Format ([SNIRF](https://github.com/fNIRS/snirf))
file specification is supported in BIDS. The SNIRF
specification supports one or more fNIRS datasets to be stored in a single
`.snirf` file. However, to be BIDS compatible, each SNIRF file MUST contain
only a single run. A limited set of fields from the SNIRF specification are
replicated  in the BIDS specification. This redundancy allows the data to be
easily parsed by humans and machines that do not have a SNIRF reader at hand,
which improves findability and tooling development.

> 

Reviewer 2: ```suggestion
Template: 

sub-<label>/
    [ses-<label>/]
        nirs/
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_nirs.snirf
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_nirs.json
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_nirs.json
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_channels.tsv
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_optodes.tsv
            sub-<label>[_ses-<label>]_task-<label>[_acq-<label>][_run-<index>][_proc-<label>]_coordsystem.json

```

We usually present a template with the data organization at the beguining

Author: Doesn't the macro in the lines above create this?

> Original unprocessed fNIRS data in the native format, if different from SNIRF, can also

Reviewer 2: ```suggestion
Raw fNIRS data in the native format, if different from SNIRF, can also
```
By original unprocessed do you mean "raw"? or does it have additional meaning? Otherwise, maybe raw is the more used way to express it throughout the specs

Author: Raw often refers to the amplifier output data in fNIRS. As opposed to optical density or haemoglobin data. So we will leave it as Original unprocessed.

### Terminology

For proper documentation of fNIRS recording metadata, it is important
to understand the difference between a Source, Detector, and Channel as these
are defined differently to other modalities, such as EEG. The following definitions
apply in this document:

-   Source - A light emitting device, sometimes called a transmitter.

-   Detector - A photoelectric transducer, sometimes called a receiver.

-   Optode - Refers to either a source or detector.

-   Channel - A paired coupling of a source and a detector with one specific wavelength of light.
    It is common for a single Source-Detector pair to result in two or more channels
    with different wavelengths.

### Sidecar JSON (`*_nirs.json`)

It is common within the fNIRS community for researchers to build their own caps
and optode holders to position their sources and detectors, or for optodes to
be directly attached to the scalp with adhesive. To facilitate description of
the wide variety of possible configurations, several fields are RECOMMENDED within
the `*_nirs.json` file.
To clarify the usage and interaction of these fields, the following examples are provided.

> If a commercial cap such as EasyCap actiCAP 64 Ch Standard-2

Reviewer 2: ```suggestion
- If a commercial cap such as EasyCap actiCAP 64 Ch Standard-2
```

> was used, then the values `CapManufacturer = ”EasyCap”`, `CapManufacturersModelName = ”actiCAP

Reviewer 2: ```suggestion
was used, then the values `CapManufacturer = "EasyCap"`, `CapManufacturersModelName = "actiCAP
```

> 64 Ch Standard-2”` and `NIRSPlacementScheme = ”n/a”` should be used.

Reviewer 2: ```suggestion
64 Ch Standard-2"` and `NIRSPlacementScheme = "n/a"` should be used.
```

> If an EasyCap was used but with custom positions,

Reviewer 2: ```suggestion
- If an EasyCap was used but with custom positions,
```

as may be done by cutting custom holes in the cap,

> then the values `CapManufacturer = ”EasyCap”`, `CapManufacturersModelName = ”custom”` and

Reviewer 2: ```suggestion
then the values `CapManufacturer = "EasyCap"`, `CapManufacturersModelName = "custom"` and
```

> `NIRSPlacementScheme = ”n/a”` should be used.

Reviewer 2: ```suggestion
`NIRSPlacementScheme = "n/a"` should be used.
```

> If a completely custom cap was knitted, then

Reviewer 2: ```suggestion
- If a completely custom cap was knitted, then
```

> `CapManufacturer = ”custom”`, `CapManufacturersModelName = ”custom”` and

Reviewer 2: ```suggestion
`CapManufacturer = "custom"`, `CapManufacturersModelName = "custom"` and
```

> `NIRSPlacementScheme = ”n/a”`.

Reviewer 2: ```suggestion
`NIRSPlacementScheme = "n/a"`.
```

> If no cap was used and optodes were taped to the scalp

Reviewer 2: ```suggestion
- If no cap was used and optodes were taped to the scalp
```

> at positions Cz, C1 and C2, then the values `CapManufacturer = ”none”`, `CapManufacturersModelName

Reviewer 2: ```suggestion
at positions Cz, C1 and C2, then the values `CapManufacturer = "none"`, `CapManufacturersModelName
```

> = ”n/a”` and `NIRSPlacementScheme = ”["Cz", "C1", “C2]”` should be used.

Reviewer 2: ```suggestion
= "n/a"` and `NIRSPlacementScheme = "["Cz", "C1", "C2"]"` should be used.
```

> 

Reviewer 2: ```suggestion
In these cases additional information regarding channels and optodes SHOULD be placed in `*_channels.tsv` and `*_optodes.tsv` files.

```
When there's no info about placement schema, etc, then this `*_channels.tsv` and `*_optodes.tsv` files SHOULD be present, right? or it's just MAY?

Closely spaced or short-separation source-detector pairs are often included in fNIRS measurements to
obtain a measure of systemic, rather than neural, activity. These source-detector

> pairs are referred to as short channels. There is variation in how manufacturers

Reviewer 2: ```suggestion
pairs are referred to as *short channels*. There is variation in how manufacturers
```

Example:

```JSON
{
  "TaskName":"visual",
  "InstitutionName":"Macquarie University. Australian Hearing Hub",
  "InstitutionAddress":"6 University Ave, Macquarie University NSW 2109 Australia",
  "Manufacturer":"NIRx",
  "ManufacturersModelName":"NIRScout",
  "TaskDescription":"visual gratings and noise patterns",
  "Instructions":"look at the dot in the center of the screen and press the button when it changes color",
  "SamplingFrequency":3.7,
  "NIRSChannelCount":56,
  "NIRSSourceOptodeCount":16,
  "NIRSDetectorOptodeCount":16,
  "ACCELChannelCount":0,
  "SoftwareFilters":"n/a",
  "RecordingDuration":233.639,
  "DCOffsetCorrection":0,
  "HardwareFilters":{"Highpass RC filter": {"Half amplitude cutoff (Hz)": 0.0159, "Roll-off": "6dBOctave"}},
  "CapManafacturer":"NIRx",
  "CapManufacturersModelName":"Headband with print (S-M)",
}
```

### Participant file

For fNIRS data, the modality agnostic participant.tsv and json files SHOULD contain

> the age field, as this is required for calculation of age-specific pathlength factors.

Reviewer 2: ```suggestion
the `age` field, as this is required for calculation of age-specific pathlength factors.
```

(for example, in the `sourcePower` field).
However, to improve searchability and ease of access for users, it is useful to
know if short channels were included in the fNIRS measurements; this information
is stored in the field `ShortChannelCount`.
If the field `ShortChannelCount` is populated, then the optional column `short_channel`
may be used in `*_channels.tsv` to describe which channels were specified as short.

Generic fields that MUST be present:

{{ MACROS___make_metadata_table(
   {
      "TaskName": "REQUIRED",
   }
) }}

Generic fields which SHOULD be present: For consistency between studies and institutions,
we encourage users to extract the values of these fields from the actual raw data.
Whenever possible, please avoid using ad hoc wording.

{{ MACROS___make_metadata_table(
   {
      "InstitutionName": "RECOMMENDED",
      "InstitutionAddress": "RECOMMENDED",
      "Manufacturer": "RECOMMENDED",
      "ManufacturersModelName": "RECOMMENDED",
      "SoftwareVersions": "RECOMMENDED",
      "TaskDescription": "RECOMMENDED",
      "Instructions": "RECOMMENDED",
      "CogAtlasID": "RECOMMENDED",
      "CogPOID": "RECOMMENDED",
      "DeviceSerialNumber": "RECOMMENDED",

>       "RecordingDuration": "RECOMMENDED",

Reviewer 3: The fields RecordingDuration, HeadCircumference, HardwareFilters, SubjectArtefactDescription are not so generic that they are shared with all modalities. It is mostly shared with EEG, partially with iEEG. I don't know whether this is a problem.

      "HeadCircumference": "RECOMMENDED",
      "HardwareFilters": "RECOMMENDED",

>       "SubjectArtefactDescription": "RECOMMENDED",

Reviewer 3: is this SubjectArtefactDescription field needed?

Reviewer 4: the drop-down also still includes an EEG specific description: 
> Freeform description of the observed subject artefact and its possible cause (for example, "Vagus Nerve Stimulator", "non-removable implant"). If this field is set to "n/a", it will be interpreted as absence of major source of artifacts except cardiac and blinks.

Reviewer 4: I think we did not draw a conclusion on this yet, as the specification was not altered so far. Should we reopen this for discussion?

| NIRSPlacementScheme | RECOMMENDED           | [string][]            | Placement scheme of NIRS optodes. Either the name of a standardized placement system (for example, "10-20") or a list of standardized position names (for example, ["Cz", "Pz"]). This field should only be used if a cap was not used. If a standard cap was used, then it should be specified in CapManufacturer and CapManufacturersModelName and this field should be set to “n/a” |

Example:

```JSON
{
  "TaskName":"visual",
  "InstitutionName":"Macquarie University. Australian Hearing Hub",
  "InstitutionAddress":"6 University Ave, Macquarie University NSW 2109 Australia",
  "Manufacturer":"NIRx",
  "ManufacturersModelName":"NIRScout",
  "TaskDescription":"visual gratings and noise patterns",
  "Instructions":"look at the dot in the center of the screen and press the button when it changes color",
  "SamplingFrequency":3.7,
  "NIRSChannelCount":56,
  "NIRSSourceOptodeCount":16,
  "NIRSDetectorOptodeCount":16,
  "ACCELChannelCount":0,
  "SoftwareFilters":"n/a",
  "RecordingDuration":233.639,

>   "DCOffsetCorrection":0,

Reviewer 3: the field DCOffsetCorrection is used in the example, but not defined in any table

Reviewer 3: the specified value is `0`, but I wonder whether it should be a JSON Boolean, i.e. `true` or `false`

  "HardwareFilters":{"Highpass RC filter": {"Half amplitude cutoff (Hz)": 0.0159, "Roll-off": "6dBOctave"}},
  "CapManafacturer":"NIRx",
  "CapManufacturersModelName":"Headband with print (S-M)",
}
```

> ### Participant file

Reviewer 3: Please insert 
The `participants.tsv` and `participants.json` files are RECOMMENDED to describe properties of participants such as age, sex, handedness. If the `participants.tsv` file exists, it MUST contain the column participant_id and SHOULD contain the `age` field ...

| NIRSCWMUA                   | Continuous wave optical absorption  measurements. Equivalent to dataTypeLabel mua in SNIRF.                                                                                  |
| ACCEL                       | Accelerometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
| GYRO                        | Gyrometer channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z).     |
| MAGN                        | Magnetomenter channel, one channel for each orientation. An extra column `component` for the axis of the orientation MUST be added to the `_*channels.tsv` file (x, y or z). |
| MISC                        | Miscellaneous                                                                                                                                                                |

Example:

```Text
Name         type                   source      detector      wavelength_nominal   units

> S1-D1        NIRSCWAMPLITUDE        A1          Fz            760                  V

Reviewer 3: how does one convert from the unit `V` (from the ADC) to the amount of light (i.e., the amount of photons) in `Lumen`, which can be converted to optical densities? In case `V` is specified, there should also be a detector-specific calibration value somewhere.

Author: In practice, just the raw voltage from the ADC is used, and no conversion to Lumen is performed. Optical density is performed on the raw ADV voltages (possibly after filtering etc).

This is discussed in  Huppert TJ, Diamond SG, Franceschini MA, Boas DA. HomER: a review of time-series analysis methods for near-infrared spectroscopy of the brain. Appl Opt. 2009 Apr 1;48(10):D280-98. doi: 10.1364/ao.48.00d280. 
> In general, the detected light intensity is proportional to the amplitude of the voltage signal reported by the photon detectors, assuming that the instability and possible nonlinearity of the photon detectors of the NIRS instrument used can be neglected. By normalizing the intensity measurements to the incident light at the start of the experiment (in practice, usually the mean of the signal over the course of the study is used), one can neglect detector efficiency, detector analog amplification (gain), and the instrument’s absolute incident laser power, since these appear as scaling factors that multiply both the numerator and denominator

However, I can imagine future uses where precise knowledge of the Lumen is useful.
I just checked the SNIRF spec and this conversion parameter does not exist (someone correct me if I missed it, I was hoping to find a consistent naming).
We could add an OPTIONAL column called `calibration_lumen` that describes `The calibration required to convert from the channel units to Lumen`. What do you think @robertoostenveld @dboas @lpollonini ?

Reviewer 3: Agreed that V (or even integer AD values) is sufficient in the subsequent analysis. I suggest not to add a calibration column to the `channels.tsv`, the calibration should be performed on the float values represented in the low-level SNIRF file. The specification of `units` here as you suggest is fine and corresponds to  [dataUnit](https://github.com/fNIRS/snirf/blob/v1.1/snirf_specification.md#nirsidatajmeasurementlistkdataunit).

Author: Thanks for the quick feedback @robertoostenveld 
I am marking this as resolved. Anyone who wants to reopen it please tag me so I get a notification (this PR is long!!)
